ops/build_containers.py

```diff
@@ -0,0 +1,96 @@
+from os import path, curdir
+from subprocess import run
+from tempfile import gettempdir, NamedTemporaryFile
+import urllib.request
+from json import loads
+
+from config import CONFIG, checkoutRepo, warpedBuildFlags
+
+dockerApiUrl="https://hub.docker.com/v2/repositories"
+
+def doesContainerExistsInCloud(name, tag):
+    try:
+        urllib.request.urlopen("{}/{}/tags/{}".format(dockerApiUrl, name, tag)).read()
+    except urllib.error.HTTPError:
+        return False
+
+    return True
+
+def getContainerInfo(name, tag):
+    if not doesContainerExistsInCloud(name, tag):
+        return {}
+
+    url = "{}/{}/tags/{}".format(dockerApiUrl, name, tag)
+    return loads(urllib.request.urlopen(url).read().decode())
+
+def compareRemoteContainers(ctr1, ctr2):
+    info1 = getContainerInfo(ctr1.split(":")[0], ctr1.split(":")[1])
+    info2 = getContainerInfo(ctr2.split(":")[0], ctr2.split(":")[1])
+    if not info1 or not info2:
+        return False
+
+    return info1['images'][0]['digest'] == info2['images'][0]['digest']
+
+def buildContainer(container, tag, dockerFile, path):
+    print("\nLOG: Building {}:{}\n".format(container, tag))
+    run([
+        "docker",
+        "build",
+        "-t",
+        "{}:{}".format(container, tag),
+        "-f",
+        dockerFile,
+        path
+    ], check=True)
+
+def reTag(container, tag1, tag2):
+    print("\nLOG: Tagging {} {} -> {}\n".format(container, tag1, tag2))
+    run([
+        "docker",
+        "tag",
+        "{}:{}".format(container, tag1),
+        "{}:{}".format(container, tag2)
+    ], check=True)
+
+def prepareDockerBuildContext(name, gitHash):
+    repoPath = path.join(gettempdir(), name)
+    if name == "authority":
+        dockerFile = path.join(repoPath, "Dockerfile.nonvoting")
+        checkoutRepo(repoPath, CONFIG["AUTH"]["REPOSITORY"], gitHash)
+    elif name == "server":
+        dockerFile = path.join(repoPath, "Dockerfile")
+        checkoutRepo(repoPath, CONFIG["SERVER"]["REPOSITORY"], gitHash)
+    elif name == "meson":
+        repoPath = curdir
+        dockerFile = "Dockerfile"
+
+    tmpdf = NamedTemporaryFile().name
+    with open(dockerFile, 'r') as infile, open(tmpdf, 'w+') as output:
+        for line in infile:
+            if "RUN cd cmd/" in line and "go build" in line and CONFIG["WARPED"]:
+                line = line.strip() + " " + warpedBuildFlags + "\n"
+            if CONFIG["SERVER"]["CONTAINER"] in line and CONFIG["WARPED"]:
+                line = "FROM {}:{}\n".format(
+                        CONFIG["SERVER"]["CONTAINER"],
+                        CONFIG["SERVER"]["TAGS"]["NAMED"],
+                    )
+
+            output.write(line)
+
+    return tmpdf, repoPath
+
+for key in ["AUTH", "SERVER", "MESON"]:
```

Comment:
Instead of having an array, you can just do `CONFIG.keys()`.

Reply:
Not all keys o CONFIG are containers. But I can create a new dict in config that has the containers.

Reply:
Restructure `CONFIG` differently.